Documentation/ApiOverview/Seo/_ExampleWebsiteTitle/_WebsiteTitleProvider.php

```diff
@@ -13,26 +13,33 @@
 #[Autoconfigure(public: true)]
 final readonly class WebsiteTitleProvider implements PageTitleProviderInterface
 {
+    protected ServerRequestInterface $request;
     public function __construct(
         private SiteFinder $siteFinder,
     ) {}
 
     public function getTitle(): string
     {
-        /** @var PageInformation $pageInformation */
-        $pageInformation = $this->getRequest()->getAttribute('frontend.page.information');
-
-        $site = $this->siteFinder->getSiteByPageId($pageInformation->getId());
+        $site = $this->siteFinder->getSiteByPageId($this->getPageInformation()->getId());
         $titles = [
-            $pageInformation->getPageRecord()['title'],
+            $this->getPageInformation()->getPageRecord()['title'] ?? '',
```

Comment:
Is null-coalescing necessary? The page record always has a title? If not, there is really something wrong which should fail.

Reply:
What do you suggest? A test if it is set and a string first?

Reply:
If you have page information, then there is a page record. And the page record has a title. I think `?? ''` is not necessary.

Reply:
$this->getPageInformation()->getPageRecord() is an array and is not promising any guaranteees beyong being an array